Plugins/Reqnroll.ExternalData/Reqnroll.ExternalData.ReqnrollPlugin/Loaders/JsonDataTableGenerator.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Newtonsoft.Json.Linq;
+using Reqnroll.ExternalData.ReqnrollPlugin.DataSources;
+
+namespace Reqnroll.ExternalData.ReqnrollPlugin.Loaders
+{
+
+    public class JsonDataTableGenerator
+    {
+        public DataSources.DataTable FlattenDataSetToDataTable(JObject originalJson, string[] objectPath)
+        {
+            try
+            {
+                var header = GenerateHeader(originalJson, objectPath);
+                var dataTable = new DataSources.DataTable(header.ToArray());
+                var records = GenerateRecordsFromNestedObjects(originalJson, objectPath, new List<DataRecord>());
+                foreach (var record in records) dataTable.Items.Add(record);
+
+                return dataTable;
+            }
+            catch (System.NullReferenceException)
```

Comment:
I would try to avoid catching NullRef exceptions as it might hide other programmer errors as well. If you need this to handle cases where the JSON structure is not complete (some nodes missing), I would do an explicit check for that in `GenerateRecordsFromNestedObjects` and throw some specific exception that you can catch and wrap.

Reply:
Agreed. As far as I can tell all the possible null reference exceptions are covered by the null checks in `GenerateRecordsFromNestedObjects` and `GenerateHeader`. I have removed the general catch of System.NullReferenceException.